From 79b7bc15eaed5030629ece3fc2502737569b1601 Mon Sep 17 00:00:00 2001
From: Sergio Durigan Junior <sergiodj@chainguard.dev>
Date: Fri, 31 Oct 2025 07:47:12 -0400
Subject: [PATCH 11/11] Add missing openssl error functions to checkVersion
 initialization

These extra functions are needed in case there's an error when loading
openssl (for example, when the configuration file is invalid).

This patch will be submitted upstream.
---
 .../github.com/golang-fips/openssl/v2/internal/ossl/zossl.c   | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/vendor/github.com/golang-fips/openssl/v2/internal/ossl/zossl.c b/src/vendor/github.com/golang-fips/openssl/v2/internal/ossl/zossl.c
index 832ecd8558..d975084e5e 100644
--- a/src/vendor/github.com/golang-fips/openssl/v2/internal/ossl/zossl.c
+++ b/src/vendor/github.com/golang-fips/openssl/v2/internal/ossl/zossl.c
@@ -678,6 +678,8 @@ void __mkcgo_load_init_3(void* handle) {
 	__mkcgo__dlsym(EVP_MD_free)
 	__mkcgo__dlsym(EVP_MD_get0_provider)
 	__mkcgo__dlsym(EVP_default_properties_is_fips_enabled)
+	__mkcgo__dlsym(ERR_get_error_all)
+	__mkcgo__dlsym(ERR_error_string_n)
 }
 
 void __mkcgo_unload_init_3() {
@@ -690,6 +692,8 @@ void __mkcgo_unload_init_3() {
 	_g_EVP_MD_free = NULL;
 	_g_EVP_MD_get0_provider = NULL;
 	_g_EVP_default_properties_is_fips_enabled = NULL;
+	_g_ERR_get_error_all = NULL;
+	_g_ERR_error_string_n = NULL;
 }
 
 void __mkcgo_load_legacy_1(void* handle) {
-- 
2.51.0

